client/layout/masterbar/logged-in.jsx

```diff
@@ -230,111 +259,218 @@ class MasterbarLoggedIn extends Component {
+						className="masterbar__item-action-search"
+						tooltip={ translate( 'Search' ) }
+						preloadSection={ this.preloadMe }
+					>
+						{ translate( 'Search Actions' ) }
+					</Item>
+				),
+			planUpsell: () =>
+				! domainOnlySite &&
+				! isMigrationInProgress && (
+					<AsyncLoad
+						require="./plan-upsell"
+						className="masterbar__item-upsell button is-primary"
+						tooltip={ translate( 'Upgrade your plan' ) }
+					>
+						{ translate( 'Upgrade' ) }
+					</AsyncLoad>
+				),
+			publish: () =>
+				! domainOnlySite &&
+				! isMigrationInProgress && (
+					<AsyncLoad
+						require="./publish"
+						placeholder={ null }
+						isActive={ this.isActive( 'post' ) }
+						className="masterbar__item-new"
+						tooltip={ translate( 'Create a New Post' ) }
+					>
+						{ translate( 'Write' ) }
+					</AsyncLoad>
+				),
+			cart: () => (
+				<AsyncLoad
+					require="./masterbar-cart/masterbar-cart-wrapper"
+					placeholder={ null }
+					goToCheckout={ this.goToCheckout }
+					onRemoveProduct={ this.onRemoveCartProduct }
+					onRemoveCoupon={ this.onRemoveCartProduct }
+					selectedSiteSlug={ currentSelectedSiteSlug }
+					selectedSiteId={ currentSelectedSiteId }
+				/>
+			),
+			me: () => (
+				<Item
+					tipTarget="me"
+					url="/me"
+					icon={ isMobile ? null : 'user-circle' }
+					onClick={ this.clickMe }
+					isActive={ this.isActive( 'me' ) }
+					className="masterbar__item-me"
+					tooltip={ translate( 'Update your profile, personal settings, and more' ) }
+					preloadSection={ this.preloadMe }
+				>
+					<Gravatar
+						className="masterbar__item-me-gravatar"
+						user={ this.props.user }
+						alt={ translate( 'My Profile' ) }
+						size={ 18 }
+					/>
+					<span className="masterbar__item-me-label">
+						{ isMobile
+							? user?.display_name
+							: translate( 'My Profile', {
+									context: 'Toolbar, must be shorter than ~12 chars',
+							  } ) }
+					</span>
+				</Item>
+			),
+			notifications: () => (
+				<Notifications
+					isShowing={ this.props.isNotificationsShowing }
+					isActive={ this.isActive( 'notifications' ) }
+					className="masterbar__item-notifications"
+					tooltip={ translate( 'Manage your notifications' ) }
+				>
+					<span className="masterbar__item-notifications-label">
+						{ translate( 'Notifications', {
+							comment: 'Toolbar, must be shorter than ~12 chars',
+						} ) }
+					</span>
+				</Notifications>
+			),
+			menu: () => (
+				<>
+					<Item
+						tipTarget="Menu"
+						icon="menu"
+						onClick={ () => this.handleToggleMenu() }
+						isActive={ this.state.isMenuOpen }
+						className="masterbar__item-menu"
+						tooltip={ translate( 'Menu' ) }
+						ref={ ( ref ) => ref !== menuBtnRef && this.setState( { menuBtnRef: ref } ) }
+					/>
+					<MasterBarMobileMenu
+						onClose={ () => this.handleToggleMenu() }
+						open={ this.state.isMenuOpen }
+					>
+						{ this.masterbarItemsMap().publish() }
```

Comment:
What's the benefit of containing each item inside an object returned from `this. masterbarItemsMap()` when compared to just having a separate method for each item? Won't it be simpler to just have a separate method for each item?